inference-engine/src/mkldnn_plugin/nodes/mkldnn_interpolate_node.cpp

```diff
@@ -0,0 +1,1950 @@
+    auto &srcMemPtr = getParentEdgeAt(DATA_ID)->getMemoryPtr();
+    auto &targetShapeMemPtr = getParentEdgeAt(TARGET_SHAPE_ID)->getMemoryPtr();
+    auto &scalesMemPtr = getParentEdgeAt(SCALES_ID)->getMemoryPtr();
+
+    uint8_t *dst_data = reinterpret_cast<uint8_t*>(dstMemPtr->GetData()) +
+            dstMemPtr->GetDescriptor().data.layout_desc.blocking.offset_padding * dstDataSize;
+    uint8_t *src_data_origin = reinterpret_cast<uint8_t*>(srcMemPtr->GetData()) +
+            srcMemPtr->GetDescriptor().data.layout_desc.blocking.offset_padding * srcDataSize;
+
+    int *targetShapeData = reinterpret_cast<int*>(targetShapeMemPtr->GetData()) +
+            targetShapeMemPtr->GetDescriptor().data.layout_desc.blocking.offset_padding;
+    int targetShapeLen = getParentEdgeAt(TARGET_SHAPE_ID)->getDesc().getDims()[0];
+    targetShape.resize(targetShapeLen);
+    for (int i = 0; i < targetShapeLen; i++) {
+        targetShape[i] = targetShapeData[i];
+    }
+
+    float *scalesData = reinterpret_cast<float*>(scalesMemPtr->GetData()) +
+            scalesMemPtr->GetDescriptor().data.layout_desc.blocking.offset_padding;
+    int scalesLen = getParentEdgeAt(SCALES_ID)->getDesc().getDims()[0];
+    scales.resize(scalesLen);
+    for (int i = 0; i < scalesLen; i++) {
+        scales[i] = scalesData[i];
+    }
+
+    SizeVector srcDim = getParentEdgeAt(DATA_ID)->getDesc().getDims();
+    size_t dimSize = srcDim.size();
+    if (isAxesSpecified) {
+        auto &axesMemPtr = getParentEdgeAt(AXES_ID)->getMemoryPtr();
+        int *axesData = reinterpret_cast<int*>(axesMemPtr->GetData()) +
+            axesMemPtr->GetDescriptor().data.layout_desc.blocking.offset_padding;
+        int axesLen = getParentEdgeAt(AXES_ID)->getDesc().getDims()[0];
+        axes.resize(axesLen);
+        for (int i = 0; i < axesLen; i++) {
+            axes[i] = axesData[i];
+        }
+    } else {
+        // default is: [0, ..., dataRank - 1]
+        axes.resize(dimSize);
+        for (int i = 0; i < dimSize; i++) {
+            axes[i] = i;
+        }
+    }
+    if (scales.size() != axes.size() || targetShape.size() != axes.size()) {
+        THROW_IE_EXCEPTION << "Interpolate layer with name '" << getName() <<
+        "' do not have the same number elements in axis as in scales or target_spatial_shape.";
+    }
+
+    SizeVector srcDimPad = getPaddedInputShape(srcDim);
+    SizeVector dstDim = outShapeCalc(srcDim);
+
+    auto to5Dim = [&](SizeVector casesDim) {
+        size_t caseSize = casesDim.size();
+        SizeVector dim5(5, 1lu);
+        if (caseSize > 2) {
+            dim5[0] = casesDim[0];
+            dim5[1] = casesDim[1];
+        }
+        if (caseSize == 5) {
+            dim5[2] = casesDim[2];
+        }
+        dim5[3] = casesDim[caseSize - 2];
+        dim5[4] = casesDim[caseSize - 1];
+        return dim5;
+    };
+    auto srcDim5d = to5Dim(srcDim);
+    auto srcDimPad5d = to5Dim(srcDimPad);
+    auto dstDim5d = to5Dim(dstDim);
```

Comment:
I would suggest to move all dims/indexes computations on initialization stage. As I understand we have two cases:
1) shapeInferMode == "sizes". 
In this case outputShape exactly equals dstDims (getParentEdgeAt(0)->getDesc().getDims()) which we know on initialization stage.
2) shapeInferMode == "scales".
 In that case you can check that `scales` input is constant. If it is constant you can extract its buffer on initialization stage as well. See for reference: https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/mkldnn_graph_optimizer.cpp#L263
For now we can throw exception about unsupported case if `scales` input is not constant.
The same applicable for 'axes' input. 
 
It should decrease number of computations on runtime stage. What do you think?

Reply:
Great idea! Dims info and index table used in execution are built offline now. thank you!